Fix providers

xhujerr opened the pull request:
- Solves issue https://github.com/puppetlabs/puppetlabs-chronos/issues/13 with pe_gem
- Chronos service is managed by upstart. I added option $service_provider to reflect that.

rji commented:
It isn't necessary to be explicit about the service provider for the Chronos service. In fact, that will break on any OSes that don't use Upstart. It's best to just let Puppet select the provider based on the underlying OS.

Also, the two commits in this PR should be squashed.

xhujerr commented:
Letting puppet select the provider was failing badly on CentOS 6.6. The default provider there is `service`, but chronos afaik requires `upstart`.

Nevertheless I agree that setting upstart as default value is bad idea. I just wanted to have an option to enforce the provider. I try to think about how to set some non-destructive default.

rji commented:
Hey @xhujerr – we haven't seen a response to our previous feedback. As I mentioned, setting a default of 'upstart' doesn't make sense for this module, and appears to be outside the scope of changing `pe-gem` to be `pe_gem`.

Please modify this PR to reflect the `pe-gem` changes only, and squash the two commits. Feel free to open a separate issue or PR to discuss the option of specifying the service provider.

xhujerr commented:
Sorry for not responding in timely manner, had a bit busy week.

I moved the service provider into https://github.com/puppetlabs/puppetlabs-chronos/issues/15 and squashed the commits.

rji commented:
This looks good to me :+1:

@Ziaunys - you might want to take a look at this too

Ziaunys commented:
+1
